x-pack/plugins/enterprise_search/public/applications/enterprise_search_content/components/search_indices/search_indices_logic.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import {
+  searchIndices as searchIndicesMock,
+  searchEngines as searchEnginesMock,
+} from '../../__mocks__';
+
+import { kea, MakeLogicType } from 'kea';
+
+import { Engine } from '../../../app_search/components/engine/types';
+import { flashAPIErrors } from '../../../shared/flash_messages';
+import { SearchIndex } from '../../types';
+
+export interface SearchIndicesValues {
+  searchIndices: SearchIndex[];
+  searchEngines: Engine[];
+}
+
+export interface SearchIndicesActions {
+  loadSearchEngines(): void;
+  onSearchEnginesLoad(searchEngines: Engine[]): Engine[]; // TODO proper types when backend ready
+  loadSearchIndices(): void;
+  onSearchIndicesLoad(searchIndices: SearchIndex[]): SearchIndex[]; // TODO proper types when backend ready
+}
+
+export const SearchIndicesLogic = kea<MakeLogicType<SearchIndicesValues, SearchIndicesActions>>({
+  path: ['enterprise_search', 'content', 'search_indices', 'search_indices_logic'],
+  actions: {
+    loadSearchIndices: true,
+    onSearchIndicesLoad: (searchIndices) => searchIndices,
```

Comment:
suggestion: I'm not a fan of this onXXX syntax in Redux/Kea actions. Redux actions are messages in a messaging bus, they are not really functions that happen and produce a result. Rather, the reducer and listener process these messages to create a new state and/or make things happen. 

The message you're sending is not that 'onLoad' is happening, it's really that the API call succeeded--so I would usually call these actions something like `searchIndicesLoadSuccess`. I'd also consider adding a `searchIndicesLoadError` action to be able to handle the error separately from the API call--makes form handling much easier, for instance.

Reply:
I will change them to `searchWhateverSuccess` since we didn't define the failure cases yet, will skip that, but definitely a nice convention to follow, I like the idea 👍

Reply:
I'm a big fan of this pattern, we should document how we build logic files for Enterprise Search